### PR TITLE
Creates code coverage github action

### PR DESCRIPTION
**Pull request by EthanHeilman**
Adds the ncruces/go-coverage-report@v0 code coverage tool to our github actions workflow. This is only run after a PR is merged to main. It is not run on every pull request.

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -56,3 +60,5 @@ jobs:
 
     - name: Test
       run: go test ./...
+
+
```
Comment (mrjoelkamp) [inline .github/workflows/go.yml:64]:
```suggestion

```

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -5,6 +5,10 @@ on:
     paths:
       - '**.go'
 
+permissions:
+  contents: write
+  pages: write
+
```
Comment (mrjoelkamp) [inline .github/workflows/go.yml:11]: what are these permissions for? I see they are required for `ncruces/go-coverage-report` in the other workflow

Comment (EthanHeilman) [reply .github/workflows/go.yml:11]: Excellent catch! I created the second workflow specifically so that the go.yml wouldn't get these permissions.

Comment (EthanHeilman) [reply .github/workflows/go.yml]: Fixed!

**Review by mrjoelkamp — APPROVED** [review]

**Pull request merged**